src/main/java/uk/gov/hmcts/reform/coh/service/AnswerStateService.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.coh.service;
+
+import javassist.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.coh.controller.exceptions.NotAValidUpdateException;
+import uk.gov.hmcts.reform.coh.domain.AnswerState;
+import uk.gov.hmcts.reform.coh.repository.AnswerStateRepository;
+
+import java.util.Optional;
+
+@Service
+@Component
+public class AnswerStateService {
+
+    private AnswerStateRepository answerStateRepository;
+
+    @Autowired
+    public AnswerStateService(AnswerStateRepository answerStateRepository) {
+        this.answerStateRepository = answerStateRepository;
+    }
+
+    public Optional<AnswerState> retrieveAnswerStateByState(String state) {
+        return answerStateRepository.findByState(state);
+    }
+
+    public boolean validateStateTransition(AnswerState sourceState, AnswerState targetState) throws NotFoundException {
+
+        Optional<AnswerState> submittedAnswerState = retrieveAnswerStateByState("SUBMITTED");
```

Comment:
Don't  understand the validation for SUBMITTED, DRAFTED and answer_edited

Reply:
Lines 30 - 41 are preamble(aka getting the state types from the db) for the actual validation of state transition which is done by from line 42+ by using a comparison between the current state (sourceState) and the proposed state (targetState).